c:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\src\AzuerApi_uqork.py:63: SAWarning: Column 'product.product_id' is marked as a member of the primary key for table 'product', but has no Python-side or server-side default generator indicated, nor does it indicate 'autoincrement=True' or 'nullable=True', and no explicit value is passed.  Primary key columns typically may not store NULL.
  session.commit()
DB Error: Instance <Product at 0x1cc41ce7740> has a NULL identity key.  If this is an auto-generated value, check that the database table allows generation of new primary key values, and that the mapped Column object is configured to expect these generated values.  Ensure also that this flush() is not occurring at an inappropriate time, such as within a load() event.
Traceback (most recent call last):
  File "c:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\src\AzuerApi_uqork.py", line 194, in <module>
    handle({"Records": [{'body': {"source_customer_id": "CUST12345","customer_name": "John Doe","submission_id": "SUBM11111","program_name": "Homeowners","product_name": "Basic Coverage","submission_status": "Open","submission_type": "New Business","effective_date": "2025-01-01T00:00:00","expiration_date": "2025-12-31T23:59:59","proposal_detail": [{"proposal_id": "QUOTE9991","proposal_status": "In Progress","proposal_effective_date": "2025-01-01T00:00:00","proposal_expiration_date": "2025-12-31T23:59:59"},{"proposal_id": "QUOTE9992","proposal_status": "Issued","proposal_effective_date": "2025-01-15T00:00:00","proposal_expiration_date": "2026-01-14T23:59:59"}]}}]}, None)
  File "c:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\src\AzuerApi_uqork.py", line 186, in handle
    consume_unqork(session, config=payload)
  File "c:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\src\AzuerApi_uqork.py", line 172, in consume_unqork
    raise e
  File "c:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\src\AzuerApi_uqork.py", line 63, in consume_unqork
    session.commit()
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 2032, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 1313, in commit
    self._prepare_impl()
  File "<string>", line 2, in _prepare_impl
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 1288, in _prepare_impl
    self.session.flush()
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4345, in flush
    self._flush(objects)
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4480, in _flush
    with util.safe_reraise():
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\util\langhelpers.py", line 224, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 4447, in _flush
    flush_context.finalize_flush_changes()
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\unitofwork.py", line 487, in finalize_flush_changes
    self.session._register_persistent(other)
  File "C:\Users\A0885080\Downloads\Lineitem_script\df_das_sqs_lambdaprocessor\venv\Lib\site-packages\sqlalchemy\orm\session.py", line 3357, in _register_persistent
    raise exc.FlushError(
sqlalchemy.orm.exc.FlushError: Instance <Product at 0x1cc41ce7740> has a NULL identity key.  If this is an auto-generated value, check that the database table allows generation of new primary key values, and that the mapped Column object is configured to expect these generated values.  Ensure also that this flush() is not occurring at an inappropriate time, such as within a load() event.
